engine/classes/ElggEntity.php

```diff
@@ -88,6 +93,14 @@
 	 * @var int
 	 */
 	protected $tables_loaded;
+
+	/**
+	 * Is the entity being saved currently? This should only be used by this class's save() method.
+	 * Subclasses that override save() don't need to use it if they call parent::save().
+	 *
+	 * @var bool
+	 */
+	protected $is_saving = false;
```

Comment:
After some thought I'm not crazy about this part of the PR. Introduces too much complexity for what we get out of it.